File: nhom6_backend/nhom6_backend/Models/Entities/Brand.cs
using System.ComponentModel.DataAnnotations;

namespace nhom6_backend.Models.Entities
{
    /// <summary>
    /// Thương hiệu sản phẩm (Gatsby, Clear, Romano, etc.)
    /// </summary>
    public class Brand : BaseEntity
    {
        /// <summary>
        /// Tên thương hiệu
        /// </summary>
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        /// <summary>
        /// Slug URL-friendly
        /// </summary>
        [MaxLength(150)]
        public string? Slug { get; set; }

        /// <summary>
        /// Mô tả thương hiệu
        /// </summary>
        [MaxLength(1000)]
        public string? Description { get; set; }

        /// <summary>
        /// Logo thương hiệu
        /// </summary>
        [MaxLength(500)]
        public string? LogoUrl { get; set; }

        /// <summary>
        /// Banner thương hiệu
        /// </summary>
        [MaxLength(500)]
        public string? BannerUrl { get; set; }

        /// <summary>
        /// Website chính thức
        /// </summary>
        [MaxLength(200)]
        public string? WebsiteUrl { get; set; }

        /// <summary>
        /// Quốc gia xuất xứ
        /// </summary>
        [MaxLength(100)]
        public string? CountryOfOrigin { get; set; }

        /// <summary>
        /// Năm thành lập
        /// </summary>
        public int? YearEstablished { get; set; }

        /// <summary>
        /// Thứ tự hiển thị
        /// </summary>
        public int DisplayOrder { get; set; } = 0;

        /// <summary>
        /// Trạng thái hiển thị
        /// </summary>
        public bool IsActive { get; set; } = true;

        /// <summary>
        /// Thương hiệu nổi bật
        /// </summary>
        public bool IsFeatured { get; set; } = false;

        /// <summary>
        /// Số lượng sản phẩm của thương hiệu
        /// </summary>
        public int ProductCount { get; set; } = 0;

        /// <summary>
        /// Danh sách sản phẩm thuộc thương hiệu
        /// </summary>
        public virtual ICollection<Product>? Products { get; set; }

        /// <summary>
        /// Meta Title cho SEO
        /// </summary>
        [MaxLength(200)]
        public string? MetaTitle { get; set; }

        /// <summary>
        /// Meta Description cho SEO
        /// </summary>
        [MaxLength(500)]
        public string? MetaDescription { get; set; }
    }
}

File: ume-backend/src/models/Brand.js
const mongoose = require('mongoose');

const brandSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  slug: { type: String, unique: true },
  description: { type: String, default: '' },
  logoUrl: { type: String, default: '' },
  bannerUrl: { type: String, default: '' },
  websiteUrl: { type: String, default: '' },
  countryOfOrigin: { type: String, default: '' },
  yearEstablished: { type: Number },
  displayOrder: { type: Number, default: 0 },
  isActive: { type: Boolean, default: true },
  isFeatured: { type: Boolean, default: false },
  productCount: { type: Number, default: 0 },
  isDeleted: { type: Boolean, default: false },
  metaTitle: { type: String, default: '' },
  metaDescription: { type: String, default: '' }
}, { timestamps: true });

brandSchema.pre('save', function(next) {
  if (this.isModified('name') && !this.slug) {
    this.slug = this.name.toLowerCase()
      .normalize('NFD').replace(/[\u0300-\u036f]/g, '')
      .replace(/đ/g, 'd').replace(/[^a-z0-9]+/g, '-').replace(/(^-|-$)/g, '');
  }
  next();
});

module.exports = mongoose.model('Brand', brandSchema);
